src/components/ha-code-editor.ts

```diff
@@ -145,28 +149,27 @@ export class HaCodeEditor extends ReactiveElement {
     this._loadedCodeMirror = await loadCodeMirror();
     const extensions: Extension[] = [
       this._loadedCodeMirror.lineNumbers(),
-      this._loadedCodeMirror.EditorState.allowMultipleSelections.of(true),
       this._loadedCodeMirror.history(),
-      this._loadedCodeMirror.highlightSelectionMatches(),
-      this._loadedCodeMirror.highlightActiveLine(),
       this._loadedCodeMirror.drawSelection(),
+      this._loadedCodeMirror.EditorState.allowMultipleSelections.of(true),
       this._loadedCodeMirror.rectangularSelection(),
+      this._loadedCodeMirror.crosshairCursor(),
+      this._loadedCodeMirror.highlightSelectionMatches(),
+      this._loadedCodeMirror.highlightActiveLine(),
       this._loadedCodeMirror.keymap.of([
         ...this._loadedCodeMirror.defaultKeymap,
         ...this._loadedCodeMirror.searchKeymap,
         ...this._loadedCodeMirror.historyKeymap,
         ...this._loadedCodeMirror.tabKeyBindings,
         saveKeyBinding,
-      ] as KeyBinding[]),
+      ]),
       this._loadedCodeMirror.langCompartment.of(this._mode),
       this._loadedCodeMirror.haTheme,
       this._loadedCodeMirror.haSyntaxHighlighting,
       this._loadedCodeMirror.readonlyCompartment.of(
         this._loadedCodeMirror.EditorView.editable.of(!this.readOnly)
       ),
-      this._loadedCodeMirror.EditorView.updateListener.of((update) =>
-        this._onUpdate(update)
-      ),
+      this._loadedCodeMirror.EditorView.updateListener.of(this._onUpdate),
```

Comment:
This should be reverted. Arrow function is required to set `this._value` in `this._onUpdate`.

Reply:
I don't understand. All this change does is make the call back defined instead of anonymous.

Reply:
Or you have to bind the context of the class by doing :
```ts
 this._loadedCodeMirror.EditorView.updateListener.of(this._onUpdate.bind(this)),
```
Otherwise, `this` will be `undefined` in `_onUpdate`.

Reply:
Oh I was confused because I had changed the definition to be an arrow function, but somehow I undid that while debugging. 🤷🏻‍♂️ 

Thanks for catching.

Reply:
Yes, arrow function works too 👍